release notes/v0.27.0.md

```diff
@@ -0,0 +1,76 @@
+- CLI: Mismatch in check counts in the end-of-test summary (#1033).
+- Config: Better validation of `stages` (#875).
+- JS: Rare panics in goja (#867).
+- HTTP: Fix request timeout and wrong context used for pushing metrics (#1260).
+
+
+## Internals
+
+- Switched to Go 1.14 for building and testing k6, bringing some fixes and performance improvements.
+- Code cleanup and formatting. Thanks @thinkerou!
+
+
+## Breaking changes
+
+- Execution config options (`scenarios`, `stages`, `iterations`, `duration`) from "upper" config layers overwrite execution options from "lower" (i.e. CLI flags > environment variables > JS options > JSON options) config layers. For example, the `--iterations` CLI flag will overwrite any execution options set as environment variables (e.g. `K6_DURATION`, `K6_STAGES`, etc.) or script options (`stages: [ /* ... */]`, `scenarios: { /* ... */ }`, etc. ).
+
+- Using different execution config options on the same level is now a configuration conflict error and will abort the script. For example, executing `k6 run --duration 10s --stages 5s:20 script.js` won't work (closes #812). The only exception is combining `duration` and `iterations`, which will result in a `shared-iterations` executor with the specified non-default `maxDuration` (#1058).
+
+- The k6 REST API for controlling script execution (i.e. the `k6 pause`, `k6 scale` commands) now only works when a `externally-controlled` executor is configured in the `scenarios` config. The initial pausing of a test (i.e. `k6 run --paused script.js`) still works with all executor types, but once the test is started with `k6 resume` (or the corresponding REST API call), it can't be paused again unless only the `externally-controlled` executor is used.
+
+- Previously, running a script with `k6 run --paused script.js` would have still executed the script's `setup()` function (if it was present and wasn't explicitly disabled with `--no-setup`) and paused immediately after. Now, k6 will pause before it executes `setup()`.
+
+- The `vusMax` / `K6_VUS_MAX` / ` -m` / `--max` option is deprecated - it was previously used for the control of the initialized VUs by the REST API. Since that has now been restricted to the `externally-controlled` executor, the equivalent option there is called `maxVUs`.
+
+- Tests with infinite duration are now only possible via the `externally-controlled` executor.
+
+- Previously, all iterations were interruptible - as soon as the specified `duration` expired, or when VUs were ramped down in a stage, any running iterations were interrupted. Now all executors besides the `externally-controlled` one have a `gracefulStop` period of `30s` by default (closes #898). Additionally, the `ramping-vus` executor has a `gracefulRampDown` parameter that configures the ramp-down grace period. For those periods, no new iterations will be started by the executors, but any currently running iterations will be allowed up to the specified periods to finish their execution.
```

Comment:
This is probably ... a bug ? (or to keep it the same as before 1007), but the `externally-controlled` executor does call gracefulStop if it's pausing vus ... 
cc @na--

Reply:
I'll fallback to @na-- on how to reword this, as I just copied it from 1007.

Reply:
hmm it's a bit tricky - the `externally-controlled` executor doesn't have `gracefulStop` as an option... but it's true that when you pause it, it calls the `vuHandle.gracefulStop()` method. This was done to preserve the logic of how "pausing" worked in the old k6, which didn't interrupt any running iterations, it just stopped k6 from making new iterations, i.e. what `gracefulStop` does: https://github.com/loadimpact/k6/blob/459da79ef51b37e5eaba4575c9065d9e592e5c49/core/local/local.go#L234-L250

so, I think the sentence here is technically correct... also, I think this whole paragraph should be more prominent, i.e. moved somewhere on top of the breaking changes, in second place perhaps?

Reply:
I moved this higher up, but maybe we should explain why the `externally-controlled` executor doesn't have the option? We mention it here and it the new docs, but it's never explained.

Since it's technically still using `vuHandle.gracefulStop()` it would make sense to be able to define this value in the config too. Avoiding special cases is always good if possible. :)

Reply:
Calling `vuHandle.gracefulStop()` is very much not the same thing as the `gracefulStop` option. It just means "don't start more iterations with this VU", whereas the option the `gracefulStop` option implies "don't start new iterations _and_ call `hardStop` after this period, if an iteration was still running".

And yes, if there's a sufficiently high demand for it, we could probably implement support for the `gracefulStop` option in the `externally-controlled` executor as well. As to why we haven't, it's because of a couple of reasons:
- the externally controlled executor is pretty much the least important executor of them all, for now, it exists mostly because of keeping _some_ backwards compatibility and because it might be useful to a small number of people
- users of the `externally-controlled` executor can achieve exactly the same result pretty easily with just the k6 REST API, if they want to - instead of directly setting the running VUs to 0, you can first run `k6 pause`, wait 30 seconds (or however long you want to give VUs to finish gracefully), and then set the active VUs to `0` - hey, presto, you have `gracefulStop` :smile: 

Though you are right, that second fact is probably worth mentioning in the docs, though probably not in the release notes

Reply:
Well, they can't have something like going down from 100 to 50 VUs gracefully ... but to be honest I don't like the `externally-controlled` executor and unless there is demand I am for *not* touching it :D